app/assets/javascripts/student_profile_v2/profile_details.js

```diff
@@ -22,55 +54,113 @@
         var score = quad[3];
         events.push({
+          type: 'MCAS ELA',
           message: name + ' scored a ' + score + ' on the ELA section of the MCAS.',
+          color: Colors.reading,
           date: QuadConverter.toDate(quad)
         });
       });
       _.each(this.props.chartData.mcas_series_math_scaled, function(quad){
         var score = quad[3];
         events.push({
+          type: 'MCAS Math',
           message: name + ' scored a ' + score + ' on the Math section of the MCAS.',
+          color: Colors.math,
           date: QuadConverter.toDate(quad)
         });
       });
       _.each(this.props.chartData.star_series_reading_percentile, function(quad){
         var score = quad[3];
         events.push({
-          message: name + ' scored a ' + score + '% on the Reading section of STAR.',
-          date: QuadConverter.toDate(quad)
+          type: 'STAR Reading',
+          message: name + ' scored in the ' + score + 'th percentile on the Reading section of STAR.',
+          date: QuadConverter.toDate(quad),
+          color: Colors.reading
         });
       });
       _.each(this.props.chartData.star_series_math_percentile, function(quad){
         var score = quad[3];
         events.push({
-          message: name + ' scored a ' + score + '% on the Math section of STAR.',
-          date: QuadConverter.toDate(quad)
+          type: 'STAR Math',
+          message: name + ' scored in the ' + score + 'th percentile on the Math section of STAR.',
+          date: QuadConverter.toDate(quad),
+          color: Colors.math
+        });
+      });
+      _.each(this.props.feed.deprecated.interventions, function(obj){
+        events.push({
+          type: 'Note',
+          message: _.template("<%=name%> (Goal: <%=goal%>)")(obj),
+          date: moment(obj.start_date_timestamp, "YYYY-MM-DD").toDate(),
+          color: Colors.procedural,
+        });
+      });
+      _.each(this.props.feed.deprecated.notes, function(obj){
+        events.push({
+          type: 'Note',
+          message: obj.content,
+          date: moment(obj.created_at_timestamp).toDate(),
+          color: Colors.procedural,
         });
       });
-      // TODO: incorporate discipline incidents.
       return _.sortBy(events, 'date').reverse();
     },
 
-    render: function() {
+    render: function(){
       return dom.div({},
-        dom.ul({},
-          this.getEvents().map(function(obj){
-            var key = [obj.date, obj.message].join();
-            return dom.li({ key: key },
-              moment(obj.date).format("MMMM Do, YYYY") + ": " + obj.message
-            );
-          })
+        dom.h4({style: styles.title}, 'Full Case History'),
+        this.renderCardList()
+      )
+    },
+
+    renderCardList: function(){
+      return dom.div({style: {marginTop: '10px'}},
+        this.getEvents().map(this.renderCard)
+      );
+    },
+
+    renderCard: function(event){
```

Comment:
For the branching here, it looks like the styling for most pieces change so it might be simpler to do two separate code paths that render the data differently.  But the ternaries seem fine to me too, up to you.